pkg/staticnetworkconfig/generator.go

```diff
@@ -137,15 +128,15 @@ func (s *StaticNetworkConfigGenerator) injectNMPolicyCaptures(hostConfig *models
 
 		// Replace logical names with capture values.
 		var modifiedInterfaceConfig string
-		// VLAN is a special case where the base-iface refers to the logical name rather than the VLAN's name.
-		if val, exists := nameToType[mac.LogicalNicName]; exists && val == "vlan" {
-			re := regexp.MustCompile(fmt.Sprintf("base-iface: %s", interfaceName))
-			modifiedInterfaceConfig = re.ReplaceAllString(interfacesWithCaptures, fmt.Sprintf(`base-iface: "{{ capture.iface%d.interfaces.0.name }}"`, j))
-		} else {
-			re := regexp.MustCompile(interfaceName)
-			modifiedInterfaceConfig = re.ReplaceAllString(interfacesWithCaptures, fmt.Sprintf(`"{{ capture.iface%d.interfaces.0.name }}"`, j))
-
-		}
+		re := regexp.MustCompile(fmt.Sprintf(`(:|-) %s($|\s)`, interfaceName))
```

Comment:
Maybe worth adding some of the edge-cases as unit-tests?
Also, a few examples in the PR description, just for context.

Reply:
I think that having multiple spaces in the yaml is valid as well e.g "- name:          eth0". 
Also, what if the interface name is surrounded with quotes e.g "- name: "eth0" or name: 'eth0'
I would unmarshal->marshal the yaml, to normalize it (I believe it will solve both of the issues I mentioned).
And as derez mentioned,  please cover this flow with unit tests (with corner cases + regular flow + the bug the PR is fixing).